packages/react/src/SecureRoute.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { FC, useEffect, useRef } from 'react';
+import { Route, RouteProps } from 'react-router-dom';
+
+import useLogto from './use-logto';
+
+const SecureRoute: FC<RouteProps> = ({ ...routeProperties }) => {
+  const { isAuthenticated, logtoClient, isLoading } = useLogto();
+  // The following useEffect takes several deps,
+  // make sure only happen when isAuthenticated is changed,
+  // prevent calling multiple times
+  const pendingLoginReference = useRef(false);
+
+  useEffect(() => {
+    const login = async () => {
+      if (pendingLoginReference.current) {
+        return;
+      }
+
+      // eslint-disable-next-line @silverhand/fp/no-mutation
+      pendingLoginReference.current = true;
+
+      await logtoClient!.loginWithRedirect(window.location.href);
```

Comment:
let's ban `foo!` in general. we can define the function after we check `logtoClient`